ui/apps/platform/src/Containers/PolicyCategories/PolicyCategoriesFilterSelect.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import { Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import { PolicyCategory } from 'types/policy.proto';
+
+type PolicyCategoriesFilterSelectProps = {
+    setCurrentPolicyCategories: (policyCategories: PolicyCategory[]) => void;
+    defaultPolicyCategories: PolicyCategory[];
+    customPolicyCategories: PolicyCategory[];
+};
+
+function PolicyCategoriesFilterSelect({
+    setCurrentPolicyCategories,
+    defaultPolicyCategories,
+    customPolicyCategories,
+}: PolicyCategoriesFilterSelectProps) {
+    const [isOpen, setIsOpen] = useState(false);
+    const [selectedFilters, setSelectedFilters] = useState<string[]>([
```

Comment:
1. What do you think about reducing responsibility for this component with `selectedFilters` and `setSelectedFilters` props in parallel with its sibling `Input` element, which has:
    * `value={filterTerm}`
    * `onChange={setFilterTerm}`
2. A string union `type Whatever = 'Default categories' | 'Custom categories'` would help TypeScript to help us make sure that all cases are covered.
3. Lifting up the responsibility to compute current policy categories according to this filter and the string filter puts all cause-and-effect in one place.
4.  For extra credit, some special care might minimize re-rendering and make it possible to use assignment to a local `currentPolicyCategories` variable instead of a `useState` hook:
     * if both of these filters are selected, then use `allPolicyCategories` instead of new array `[...defaultPolicyCategories, ...customPolicyCategories]`
     * if no string filter, skip `filter` method instead of new array